test/functionalTest/cases/3119_qc_nullcharacter_payload/qc_nullcharacter_payload.test

```diff
@@ -0,0 +1,71 @@
+# Copyright 2018 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Inappropriate parameter in query context payload.
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. Request context query with inappropriate value parameter in payload.
+#
+echo "01. Request context query with inappropriate value parameter in payload."
+echo "========================================================================"
+payload='{
+    "entities": [
+        {
+            "type": "Room",
+            "isPattern": "true",
+            "id": "\u0000"
+        }
+    ]
+}'
+orionCurl --url '/v1/queryContext' --payload "$payload"
+echo
+echo
+
+
+--REGEXPECT--
+01. Request context query with inappropriate value parameter in payload.
+========================================================================
+HTTP/1.1 200 OK
+Content-Length: 118
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "errorCode": {
+        "code": "400",
+        "details": "invalid regex for entity id pattern",
+        "reasonPhrase": "Bad Request"
+    }
+}
+
```

Comment:
I have run this .test in my environment I'm I get a different result:

```
01. Request context query with inappropriate value parameter in payload.
========================================================================
HTTP/1.1 200 OK
Content-Length: 105
Content-Type: application/json
Fiware-Correlator: 56455064-261d-11e8-8333-000c29173617
Date: Mon, 12 Mar 2018 17:46:47 GMT

{
    "errorCode": {
        "code": "400",
        "details": "invalid regex for entity id pattern",
        "reasonPhrase": "Bad Request"
    }
}
```

Most the same, but Content-Lenght differs.

Could you double-check, please? (it could be something wrong in my setup)

Reply:
@fgalan Done. Updated it to 136.
Got successful result
```
make[1]: Leaving directory `/root/fiware-orion/BUILD_RELEASE'
./test/functionalTest/testHarness.sh
Mon Mar 12 13:27:12 EDT 2018
0001/1: 3119_qc_nullcharacter_payload/3119_qc_nullcharacter_payload.test .............................................  01 seconds
```

Reply:
I'm a bit surprissed...

In my case, it worked with Content-Lenght: 105. In you case it worked with Content-Lengh: 136. I don't know how to explain this fact...

But let's merge and let jenkins CI to check. In the worse case, the PR will be reverted and we could re-examine the case.